checkov/openapi/checks/resource/v2/OperationObjectBasicAuth.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from typing import Any, Union, List
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.common.checks.enums import BlockType
+from checkov.openapi.checks.resource.v2.BaseOpenapiCheckV2 import BaseOpenapiCheckV2
+
+
+class OperationObjectBasicAuth(BaseOpenapiCheckV2):
+    def __init__(self) -> None:
+        id = "CKV_OPENAPI_15"
+        name = "Ensure that operation objects do not use basic auth - version 2.0 files"
+        categories = [CheckCategories.API_SECURITY]
+        supported_resources = ["paths"]
+        super().__init__(
+            name=name,
+            id=id,
+            categories=categories,
+            supported_entities=supported_resources,
+            block_type=BlockType.DOCUMENT,
+        )
+
+    def scan_openapi_conf(  # type:ignore[override]
+            self, conf: dict[str, Any], entity_type: str
+    ) -> tuple[CheckResult, Union[dict[str, Any], List[Any]]]:
+        paths = conf.get('paths', {}) or {}
+        security_definitions = conf.get('securityDefinitions', {}) or {}
+
+        for path, path_dict in paths.items():
+            if self.is_start_end_line(path):
+                continue
+            for operation, operation_dict in path_dict.items():
+                if self.is_start_end_line(operation):
+                    continue
+                security = operation_dict.get('security', [])
+                for security_definition in security:
+                    for auth_key in security_definition:
```

Comment:
```suggestion
                    for auth_key, auth_definition in security_definition.items():
```
and check before, if `security_definition` is a `dict`

Reply:
I take the `auth_definition` from `securitt_definitions` and not from `security_definition` so not relevant here